src/mlpack/core/tree/rectangle_tree/rectangle_tree.hpp

```diff
@@ -91,8 +93,8 @@ class RectangleTree
   std::vector<size_t> points;
   //! The local dataset
   MatType* localDataset;
-  //! The class that performs the split of the node.
-  SplitType<RectangleTree> split;
+  //! A tree-specific information
+  AuxiliaryInformationType<RectangleTree> *auxiliaryInfo;
```

Comment:
Are you sure we want to allocate this on the heap?  We then have to manage the memory and make sure it is valid.  I think usually the `AuxiliaryInformationType` classes will be small (looks like the one for the Hilbert R tree will be maybe 32 bytes), so I don't know if this is worth doing.  (A few more comments about the AuxiliaryInformationType are coming but I have to step out, so it may be a little while until I am able to write those comments.)